### PR TITLE
deps: fix dependency build

### PR DESCRIPTION
**Pull request by kolea2**
Towards #2296

### CODE REVIEW

**Comment by kolea2** [comment]
still need to annotate these with comments, just testing with build

**Comment by kolea2** [comment]
Ready for review, PTAL

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 1]
```diff
@@ -48,17 +68,57 @@ limitations under the License.
...
       <groupId>org.apache.beam</groupId>
-      <artifactId>beam-sdks-java-io-hadoop-common</artifactId>
+      <artifactId>beam-sdks-java-core</artifactId>
       <version>${beam.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
+    </dependency>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml]: we shouldnt have both commons-logging & slf4j. We should pick one and remove the others.

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 2]
```diff
@@ -48,17 +68,57 @@ limitations under the License.
...
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-hbase-1.x-shaded</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud.bigtable</groupId>
+      <artifactId>bigtable-client-core</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml]: Please keep some sort of logically ordering of dependencies...it'll be really hard to maintain like dependencies sprinkled throughout the file. Also please be consistent with `com.google.cloud.bigtable` vs `${project.groupId}`

File: bigtable-dataflow-parent/bigtable-beam-import/pom.xml [hunk 3]
```diff
@@ -87,35 +147,35 @@ limitations under the License.
...
+    </dependency>
   </dependencies>
 
   <build>
     <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <usedDependencies>commons-codec:commons-codec,com.squareup.okhttp:okhttp,io.grpc:grpc-netty-shaded,io.grpc:grpc-all,io.grpc:grpc-core,org.apache.beam:beam-runners-direct-java</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-beam-import/pom.xml]: I don't think all of these should be here. I would expect to only see org.apache.beam:beam-runners-direct-java

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 4]
```diff
@@ -70,22 +70,31 @@ limitations under the License.
...
-          <version>${hamcrest.version}</version>
-          <scope>test</scope>
+            <groupId>com.google.api</groupId>
+            <artifactId>api-common</artifactId>
+            <version>1.7.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-shaded-client</artifactId>
+            <version>1.2.6</version>
```
Comment (igorbernstein2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml]: Shouldnt this use `${hbase.version}`?

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-hadoop/pom.xml [hunk 5]
```diff
@@ -116,6 +131,14 @@ limitations under the License.
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <usedDependencies>com.google.code.findbugs:jsr305,commons-logging:commons-logging,io.dropwizard.metrics:metrics-core</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-hadoop/pom.xml]: I dont quite understand why these are here

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 6]
```diff
@@ -288,6 +288,36 @@ limitations under the License.
             <artifactId>hbase-client</artifactId>
             <version>${hbase.version}</version>
         </dependency>
+        <dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: all of these should be using the test scope

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 7]
```diff
@@ -288,6 +288,36 @@ limitations under the License.
...
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-common</artifactId>
+            <version>${hbase.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: please maintain some sort of logical ordering

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml [hunk 8]
```diff
@@ -305,31 +335,13 @@ limitations under the License.
...
-            <artifactId>hbase-server</artifactId>
-            <version>${hbase.version}</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
-            <groupId>org.apache.hbase</groupId>
-            <artifactId>hbase-testing-util</artifactId>
-            <version>${hbase.version}</version>
-            <scope>test</scope>
-        </dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-integration-tests/pom.xml]: This is a bit surprising....these are necessary to run the minicluster tests

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 9]
```diff
@@ -54,13 +54,46 @@ limitations under the License.
...
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
+            <artifactId>hadoop-mapreduce-client-core</artifactId>
+            <version>${hadoop.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-common</artifactId>
+            <version>2.7.4</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: `${hadoop.version}`

Comment (igorbernstein2) [reply bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: also, should be provided scope

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 10]
```diff
@@ -54,13 +54,46 @@ limitations under the License.
...
+        <dependency>
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-annotations</artifactId>
+            <version>${hbase.version}</version>
+        </dependency>
         <dependency>
             <groupId>org.apache.hadoop</groupId>
-            <artifactId>hadoop-client</artifactId>
+            <artifactId>hadoop-mapreduce-client-core</artifactId>
+            <version>${hadoop.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: provided scope

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 11]
```diff
@@ -54,13 +54,46 @@ limitations under the License.
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hbase</groupId>
+            <artifactId>hbase-client</artifactId>
+            <version>${hbase.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-annotations</artifactId>
             <version>${hadoop.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: provided scope

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml [hunk 12]
```diff
@@ -54,13 +54,46 @@ limitations under the License.
...
-            <scope>provided</scope>
         </dependency>
-
         <dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/pom.xml]: please maintain some sort of logical ordering

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x-shaded/pom.xml [hunk 13]
```diff
@@ -280,6 +280,13 @@ limitations under the License.
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <usedDependencies>com.google.cloud.bigtable:bigtable-hbase-1.x,org.apache.hbase:hbase-shaded-client,com.google.code.findbugs:jsr305,commons-logging:commons-logging,io.dropwizard.metrics:metrics-core,org.checkerframework:checker-compat-qual,io.opencensus:opencensus-impl,io.opencensus:opencensus-contrib-zpages,io.opencensus:opencensus-exporter-trace-stackdriver</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x-shaded/pom.xml]: This list is a bit surprising. I think we should go over it together

File: bigtable-hbase-1.x-parent/bigtable-hbase-1.x/pom.xml [hunk 14]
```diff
@@ -39,10 +39,51 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>proto-google-cloud-bigtable-admin-v2</artifactId>
+      <version>0.74.0</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-1.x-parent/bigtable-hbase-1.x/pom.xml]: This version should be inherited from elsewhere

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml [hunk 15]
```diff
@@ -75,6 +75,21 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>2.8.5</version>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>${junit.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml]: junit shouldnt be a dependency

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml [hunk 16]
```diff
@@ -75,6 +75,21 @@ limitations under the License.
...
+    <dependency>
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase-common</artifactId>
+      <version>${hbase.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-common</artifactId>
+      <version>2.8.5</version>
+    </dependency>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-hadoop/pom.xml]: I'm a bit surprised to see these here. Why are they necessary?

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml [hunk 17]
```diff
@@ -317,13 +317,46 @@ limitations under the License.
...
+        <dependency>
+            <groupId>commons-lang</groupId>
+            <artifactId>commons-lang</artifactId>
+            <version>${commons-lang.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.google.code.findbugs</groupId>
+            <artifactId>jsr305</artifactId>
+            <version>${jsr305.version}</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: scope test for everything

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml [hunk 18]
```diff
@@ -375,6 +408,13 @@ limitations under the License.
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
+                    <usedDependencies>org.hamcrest:hamcrest-core</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml]: Thats really odd

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x-shaded/pom.xml [hunk 19]
```diff
@@ -298,6 +298,13 @@ limitations under the License.
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <configuration>
+          <usedDependencies>com.google.cloud.bigtable:bigtable-hbase-2.x,org.apache.hbase:hbase-shaded-client,com.google.code.findbugs:jsr305,commons-logging:commons-logging,io.dropwizard.metrics:metrics-core,org.checkerframework:checker-compat-qual,io.opencensus:opencensus-impl,io.opencensus:opencensus-contrib-zpages,io.opencensus:opencensus-exporter-trace-stackdriver</usedDependencies>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x-shaded/pom.xml]: we should go over this together

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml [hunk 20]
```diff
@@ -36,25 +36,81 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>proto-google-cloud-bigtable-admin-v2</artifactId>
+      <version>0.74.0</version>
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>3.1.2</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml]: should use a variable

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml [hunk 21]
```diff
@@ -36,25 +36,81 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <version>${guava.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>proto-google-cloud-bigtable-admin-v2</artifactId>
+      <version>0.74.0</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml]: should use a variable

File: bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml [hunk 22]
```diff
@@ -36,25 +36,81 @@ limitations under the License.
...
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+      <version>3.1.2</version>
+    </dependency>
+    <dependency>
+      <groupId>com.google.cloud</groupId>
+      <artifactId>google-cloud-bigtable</artifactId>
+      <version>0.109.0</version>
```
Comment (igorbernstein2) [inline bigtable-hbase-2.x-parent/bigtable-hbase-2.x/pom.xml]: should use a variable

File: bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml [hunk 23]
```diff
@@ -205,6 +220,13 @@ limitations under the License.
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
```
Comment (kolea2) [inline bigtable-dataflow-parent/bigtable-hbase-beam/pom.xml]: add comment

**Pull request closed without merging**